controllers/solrprometheusexporter_controller_test.go

```diff
@@ -656,10 +657,14 @@ func TestMetricsReconcileWithUserProvidedConfig(t *testing.T) {
 	updatedConfigXml := "<config>updated by user</config>"
 	updateUserProvidedConfigMap(testClient, g, userProvidedConfigMapNN, map[string]string{util.PrometheusExporterConfigMapKey: updatedConfigXml})
 
-	// reconcile should happen again
+	// reconcile should happen again 2x
 	g.Eventually(requests, timeout).Should(gomega.Receive(gomega.Equal(expectedMetricsRequest)))
+	g.Eventually(requests, timeout).Should(gomega.Receive(gomega.Equal(expectedMetricsRequest)))
+
+	deployment = &appsv1.Deployment{}
+	err = testClient.Get(context.TODO(), metricsDKey, deployment)
+	g.Expect(err).NotTo(gomega.HaveOccurred())
 
-	deployment = expectDeployment(t, g, requests, expectedMetricsRequest, metricsDKey, userProvidedConfigMap.Name)
```

Comment:
Was `expectDeployment` not giving you the updated deployment?

Reply:
that was me just trying to simplify test logic but I changed it back